edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/IEdgeHub.cs

```diff
@@ -29,8 +29,8 @@ public interface IEdgeHub : IDisposable
 
         Task SendC2DMessageAsync(string id, IMessage message);
 
-        Task AddSubscription(string id, DeviceSubscription deviceSubscription);
+        Task ProcessSubscription(string id, DeviceSubscription deviceSubscription, bool addSubscription);
```

Comment:
I hate to do this, but maybe it makes more sense to keep the `Add`/`RemoveSubscription`. This makes it more explicit without having an opaque boolean define if the subscription is added or removed.